### PR TITLE
remote-desktop: add the ability to communicate via an EIS socket

### PR DESCRIPTION
**Pull request by whot**
This is intended as replacement for the `NotifyFoo` methods. libei
provides a more flexible and powerful method of sending input events to
the compositor.

A new method `ConnectToEIS` requests a file descriptor from the compositor
which can then be plugged into libei.

Once established, the communication between compositor and application
is direct, without the need to go through the portal process(es).

To avoid ambiguities between NotifyFoo and input events sent via libei,
any application that uses an EIS connection cannot use the NotifyFoo
methods.

One notable comment: I decided to pass the EIS connection down through the `impl.portal` instead of pipewire's approach of having the portal connect directly. Unlike the single pipewire daemon, an EIS implementation can be provided by multiple processes, so I figured it's easier to pass this down as close as possible to the entity that knows who the EIS implementation 

Eventually (and once libreis is ready) this will have the hooks to auto-set the app-id on the EIS connection as well though strictly speaking it's not necessary since the app-id is passed down to whoever needs to set up the socket anyway.

Noteworthy: there is no keysym event in libei at this point, so there's no equivalent to NotifyKeyboardKeysym

cc @jadahl

### CODE REVIEW

**Comment by whot** [comment]
ftr, we have a working (though not yet complete) use-case in this [input-leap PR](https://github.com/input-leap/input-leap/pull/1524)

File: src/remote-desktop.c
Comment (salmanmlk) [inline src/remote-desktop.c:1462]: Will it make sense to do this only if the backend call succeeded so that we can allow the existing/slower input injection implementation as the fallback?

Comment (whot) [reply src/remote-desktop.c:1460]: yes, good point, I'll add that to my list

Comment (whot) [reply src/remote-desktop.c:1461]: done in the lastest commit now.

**Comment by whot** [comment]
Marking this as ready. There's no immediate libei dependency other than the socket being passed, so even if libei takes another few weeks for the official "it's stable" API this bit is ready.

Squashed the commits together, there is no purpose keeping the various bugfixes we had in the git history after the merge.

**Comment by whot** [comment]
ftr, libei is now declared stable. Not that it matters for this PR since we don't actually do anything with libei directly.

File: tests/test_remotedesktop.py
Comment (jadahl) [inline tests/test_remotedesktop.py:171]: A potential extra test is

```python
        with self.assertRaises(dbus.exceptions.DBusException) as cm:
            fd = rd_intf.NotifyKeyboardKeycode(
                session.handle, dbus.Dictionary({}, signature="sv", 30, 1)
            )
        assert "Purposely failing to use old event notification" in cm.exception.get_dbus_message()
```

File: data/org.freedesktop.portal.RemoteDesktop.xml
Comment (whot) [reply data/org.freedesktop.portal.RemoteDesktop.xml:26]: git rebase to the rescue!

File: data/org.freedesktop.portal.RemoteDesktop.xml
Comment (whot) [reply data/org.freedesktop.portal.RemoteDesktop.xml]: fixed, thanks

File: tests/test_remotedesktop.py
Comment (whot) [reply tests/test_remotedesktop.py:167]: Added a terst for all the `NotifyFoo` funcs to return an error. fwiw, I'm not super-happy with the error message but I'd rather that minor bit doesn't hold up this PR

File: tests/test_remotedesktop.py
Comment (jadahl) [inline tests/test_remotedesktop.py:229]: Maybe the error should be tweaked to be less specific, e.g. just  "Not allowed"?

**Review by jadahl — APPROVED** [review]
All lgtm, I think this is ready to land. Can consider tweaking the error message, but nut super urgent.

File: tests/test_remotedesktop.py
Comment (whot) [reply tests/test_remotedesktop.py]: Follow-up commit changes this to `Session is not allowed to call NotifyPointer methods` and the same for `Keyboard` and `Touch`

File: data/org.freedesktop.impl.portal.RemoteDesktop.xml
Comment (GeorgesStavracas) [inline data/org.freedesktop.impl.portal.RemoteDesktop.xml:363]: Adding new methods require bumping the interface version, I think - so that apps can then decide whether or not to call this method solely based on the version number.

File: src/remote-desktop.c
Comment (GeorgesStavracas) [inline src/remote-desktop.c:1479]: Please remove this newline

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]

File: data/org.freedesktop.impl.portal.RemoteDesktop.xml
Comment (jadahl) [reply data/org.freedesktop.impl.portal.RemoteDesktop.xml:363]: The version was bumped by the clipboard remote desktop integration pull request.

Comment (GeorgesStavracas) [reply data/org.freedesktop.impl.portal.RemoteDesktop.xml:363]: Ah my bad then.

File: src/remote-desktop.c
Comment (GeorgesStavracas) [reply src/remote-desktop.c:1479]: Let's not block on this

**Review by GeorgesStavracas — APPROVED** [review]

File: src/remote-desktop.c
Comment (whot) [reply src/remote-desktop.c:1479]: ftr, newline is there to maintain existing grouping, `create/select/start` are in one group and the various `notify` functions in another. Happy to remove it if you want me to though.

Comment (GeorgesStavracas) [reply src/remote-desktop.c:1479]: Nah, it's all good.

**Pull request merged**
